docs/platform_developers/developer_getting_started.rst

```diff
@@ -0,0 +1,118 @@
+
+In order to install VS Code, you can refer to `this <https://code.visualstudio.com/learn/get-started/basics>`_ page.
+
+Further information about Inmanta VS Code extension is available on `this <https://github.com/inmanta/vscode-inmanta>`_ page.
+
+
+Setting up Python virtual environments
+########################################
+
+Python virtual environments have to be created using ``venv`` by running:
+
+.. code-block:: bash
+    
+    python3 -m venv ~/.virtualenvs/<env_name>
+
+Then activated by running:
+
+.. code-block:: bash
+    
+    source ~/.virtualenvs/<env_name>/bin/activate
+
+**Upgrading your ``pip`` will save you a lot of time and troubleshooting (due to changes in the pip resolver in version 20 and 21).** you can do so by running:
+
+.. code-block:: bash
+    
+    pip install --upgrade pip
+
+Usually the projects that you will work on, come with ``requirements.txt`` or ``requirements.dev.txt`` to install the required modules, if not, it is a good idea to ``pip install``:
+
+1. inmanta-core
+2. pytest
+3. pytest-inmanta
+4. pytest-inmanta-* (Additional pytest fixtures exist for working with yang and lsm, but are only useful when these are used in your project or module.)
+
+
+Benefit from linting and code navigation by setting up a project
+##################################################################
+
+At the time of this writing, linting and code navigation in IDEs work only if you have a project, so if you want to work on a single module, you still need to have a project.
+
+There are two things that can be done:
+
+1. Either there is already a project and you ``git clone`` your module there.
+2. Make a project.
+
+Steps to create a project are mentioned `here <https://docs.inmanta.com/community/latest/model_developers/configurationmodel.html>`_ for further reading.
+
+Once you are done with creating a project, you can ``cd`` into that directory and open vs code by running:
+
+.. code-block:: bash
+    
+    code .
+
+Upon opening your vs code, and the ``main.cf`` file, you should see modules downloading in ``libs`` directory.
+
+
+Set project sources
+#####################
+
+The next step is to set the sources of your project so that it knows, where to get its required modules from.
+
+1. Find the module you want to work on
+2. Copy the SSH URL of the repo
+3. In your vs code, open the ``project.yml`` file and under ``repo:``, add the copied line there but keep in mind to replace the name of a specific module with a place holder, like below example:
+
+.. code-block:: yaml
+    
+    repo:
+        - git@code.inmanta.com:example/my_module.git
+
+To:
+
+.. code-block:: yaml
+    
+    repo:
+        - git@code.inmanta.com:example/{}.git
+
+Now, in your ``main.cf`` file, if you import a module like, ``import nokia_service_vprn`` and save the file, you can get code completion.
+
+**Please note, code completion and navigation work on modules that are imported in the ``main.cf`` file**
+
+
+Module developers guide
```

Comment:
I would add the option here: when starting a new module, this, otherwise that.

Reply:
Done